src/res/flame.html

```diff
@@ -262,26 +559,76 @@ <h1>/*title:*/</h1>
 		search(false);
 	}
 
+	btnInvert.onclick = function(e) {
+		invertMode();
+		e.preventDefault();
+	}
+
+	btnInvert.dblclick = function(e) {
+		e.preventDefault();
+	}
+
+	btnInvert.onmousedown = function(e) {
+		e.preventDefault();
+		e.stopPropagation();
+	}
+
+	document.onclick = function () {
+		ctxMenu.style.display = 'none';
+	}
+
+	ctxResetHighlight.onclick = function(e) {
+		e.stopPropagation();
+		focusOnType = null;
+		render(root);
+		ctxMenu.style.display = 'none';
+	}
+
 	window.onkeydown = function(event) {
-		if ((event.ctrlKey || event.metaKey) && event.key === 'f') {
-			event.preventDefault();
-			search(true);
-		} else if (event.key === 'Escape') {
+		if (event.key === 'Escape') {
 			search(false);
-		} else if ((event.key === 'n' || event.key === 'N') && nav.length > 0) {
-			navIndex = (navIndex + (event.shiftKey ? nav.length - 1 : 1)) % nav.length;
-			render(nav[navIndex]);
-			document.getElementById('matchval').textContent = matchval + ' (' + (navIndex + 1) + ' of ' + nav.length + ')';
-			window.scroll(0, inverted ? root.level * 16 : canvasHeight - (root.level + 1) * 16);
-			canvas.onmousemove();
+		} else if(event.ctrlKey || event.metaKey) {
+			switch(event.key) {
+				case 'f':
+					event.preventDefault();
+					search(true);
+					break;
+			}
+		} else {
+			switch(event.key) {
+				case 'n':
+					if (nav.length > 0) {
+						navIndex = (navIndex + (event.shiftKey ? nav.length - 1 : 1)) % nav.length;
+						render(nav[navIndex]);
+						document.getElementById('matchval').textContent = matchval + ' (' + (navIndex + 1) + ' of ' + nav.length + ')';
+						window.scroll(0, inverted ? root.level * 16 : canvasHeight - (root.level + 1) * 16);
+						canvas.onmousemove();
+					}
+					break;
+				case 'i':
+					event.preventDefault();
+					invertMode();
+					break;
+				case '0':
+					event.preventDefault();
+					search(false, true);
+					break;
+			}
 		}
 	}
 
+	function invertMode() {
+		inverted = !inverted;
+		btnInvert.style.rotate = inverted ? '180deg' : '0deg';
```

Comment:
nit: the icon is _rotated_, but the graph is _mirrored_, not the same thing :)

Reply:
But seriously, I'd prefer a static icon (like up-down triangles or whatever you invent). Two-state toggle buttons are always confusing: whether they show current state or a state to switch to.